### ISSUE TITLE
select_related() not referencing correctly

### ISSUE THREAD

**Issue by troygrosfield** [hmarr/mongoengine#462]
I have a issue where select_related is incorrectly referencing documents when using a DictField.  Example:

class Example(AbstractBaseDocument):
    """
    shares = dict keyed by user id.  Each dictionary contains the following values:
                   x = some value
                   y = some value
                   z = some value
    """
    created_user = ReferenceField(User)
    shares = DictField()

How it's stored (correctly) in mongo:

{
    "shares" : {
        "357m7Vs2GY" : {
            "x" : "Some Value",
            "y" : "Some Value",
            "z" : "Some Value"
        },
                "fdskh29JK" : {
            "x" : "Some Value",
            "y" : "Some Value",
            "z" : "Some Value"
        }
    },
    "created_user" : {
        "$ref" : "user",
        "$id" : "357m7Vs2GY"
    }
}

shares is a dictionary keyed by a user id. So, when select_related() method is called, the value of shares is a user object, not the expected x, y, z dictionary key/value pairs.  It's in the _attach_objects method in the DeReference class in mongoengine/dereference.py.  I'm using MongoEngine v0.6.2.

**Comment by rozza** [comment]
I cant duplicate - can you provide a test case?

This works with the current master (soon to be 0.6.8)

```
import unittest
from mongoengine import *


class BugFixTest(unittest.TestCase):


    def setUp(self):
        conn = connect(db='mongoenginetest')

    def test_example(self):

        class User(Document):
            name = StringField()

        class Example(Document):
            """
            shares = dict keyed by user id. Each dictionary contains the following values:
            x = some value
            y = some value
            z = some value
            """
            created_user = ReferenceField(User)
            shares = DictField()

        User.drop_collection()
        Example.drop_collection()

        u = User(name="Ross").save()
        shares = {str(u.id): {'x': 1, 'y': 2}}
        Example(created_user=u, shares=shares).save()
        example = Example.objects.select_related(10)[0]
        self.assertEqual(example.shares, shares)
```

**Comment by rozza** [comment]
It might have been fixed in the releases post 0.6.2 :)
